src/kernel-install/kernel-install

```diff
@@ -90,10 +90,18 @@ if [[ ! $COMMAND ]] || [[ ! $KERNEL_VERSION ]]; then
     exit 1
 fi
 
-[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && [ "$(stat -fc %T /etc/machine-id)" != "tmpfs" ] && read -r MACHINE_ID < /etc/machine-id
+# Prefer to use an existing machine ID from /etc/machine-info or /etc/machine-id. If we're using the machine
+# ID /etc/machine-id, try to persist it in /etc/machine-info. If no machine ID is found, try to generate
+# a new machine ID in /etc/machine-info. If that fails, use "Default".
+
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && read -r MACHINE_ID </etc/machine-id
+[ -n "$MACHINE_ID" ] && [ -z "$KERNEL_INSTALL_MACHINE_ID" ] && echo "KERNEL_INSTALL_MACHINE_ID=$MACHINE_ID" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && NEW_MACHINE_ID="$(systemd-id128 new)" && echo "KERNEL_INSTALL_MACHINE_ID=$NEW_MACHINE_ID" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
```

Comment:
Wouldn't it make sense at this point to write this new MACHINE_ID also in /etc/machine-id, if it's writable? So that they are in sync?

Reply:
(because if this branch executes, then it means /etc/machine-id is empty, right?)

Reply:
Yeah, but the idea is that it doesn't really matter if they get out of sync, we just need a unique UUID so we don't cause conflicts, it could literally be anything. So since there's no reason to have them in sync, it seems better to not touch the machine ID at all.

Reply:
The problem is that sometimes they will match, and sometimes they will not, which at least I find confusing. If there was any then yeah it should be left alone, but if it's not set, why not do it? If it's writable, it just means that a different uuid will be created at some point later for it anyway, no?

Reply:
In particular, when creating a "golden image" creation of `/etc/machine-id` would be unwanted. So would say that creation of that file is not the responsibility of kernel-install.